components/places/src/db/schema.rs

```diff
@@ -113,6 +122,15 @@ const CREATE_TRIGGER_AFTER_INSERT_ON_PLACES: &str = "
     END
 ";
 
+const CREATE_TRIGGER_MOZPLACES_AFTERDELETE: &str = "
+    CREATE TEMP TRIGGER moz_places_afterdelete_trigger
+    AFTER DELETE ON moz_places
+    FOR EACH ROW WHEN OLD.sync_status == 2 -- 2 == SyncStatus::Normal
```

Comment:
Oh, clever! This has the nice side effect of automatically storing incoming synced tombstones.